src/main/java/duke/commands/FindCommand.java

```diff
@@ -0,0 +1,34 @@
+package duke.commands;
+
+import duke.exceptions.InvalidTaskIndexException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+
+/**
+ * Command to find a task from a given string prefix.
+ */
+public class FindCommand extends Command {
+    private String prefix;
+
+    public FindCommand(String prefix) {
+        this.prefix = prefix;
+    }
+
+    @Override
+    public String execute(TaskList taskList, Storage storage) {
+        int count = 0;
+        try {
+            String output = "Here are the matching tasks in your list:\n";
+            for (int i = 0; i < taskList.size(); i++) {
+                if (taskList.getTask(i).getTaskName().matches("(.*)" + this.prefix + "(.*)")) {
+                    output += (count + 1) + "." + taskList.getTask(i) + "\n";
+                    count++;
+                }
+            }
+            return output;
+        } catch (InvalidTaskIndexException ex) {
+            return "Something went wrong.";
```

Comment:
Maybe add something more descriptive so as to let users know the error in their input.

Reply:
Yup agreed! Perhaps you can also add in the correct format that the user is suppose to enter so that they have a point of reference 😄😄